libs/api/src/lib/graphql/action/action.private-queries.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Author,
+  Article,
+  Comment,
+  Page,
+  Poll,
+  PrismaClient,
+  Subscription,
+  User,
+  Event
+} from '@prisma/client'
+import {Action, ActionType} from '../../db/action'
+import {Context} from '../../context'
+import {
+  CanGetArticles,
+  CanGetAuthors,
+  CanGetComments,
+  CanGetEvent,
+  CanGetPages,
+  CanGetPoll,
+  CanGetSubscriptions,
+  CanGetUsers
+} from '@wepublish/permissions/api'
+import {hasPermission} from '@wepublish/permissions/api'
+
+export const getActions = async (
+  authenticate: Context['authenticate'],
+  article: PrismaClient['article'],
+  page: PrismaClient['page'],
+  comment: PrismaClient['comment'],
+  subscription: PrismaClient['subscription'],
+  author: PrismaClient['author'],
+  poll: PrismaClient['poll'],
+  user: PrismaClient['user'],
+  event: PrismaClient['event']
+): Promise<Array<Action>> => {
+  const {roles} = authenticate()
+
+  const articles = hasPermission(CanGetArticles, roles)
```

Comment:
Currently all of these are loaded after each other while we can load them all at the same time.

Reply:
I hope this works better now? Although not sure how much is necessary to optimize since anyhow this will be changed after actions table is added.